### PR TITLE
[Insights] Using SDK 20.1-preview with some fixes. Fixing reported issues with the Diagnostic Settings cmdlets #6998

### PR DESCRIPTION
**Pull request by markcowl**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Insights/Commands.Insights.Test/ScenarioTests/ActionGroupsTests.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public ActionGroupsTests(Xunit.Abstractions.ITestOutputHelper output)
             XunitTracingInterceptor.AddToContext(_logger);
         }
 
-        [Fact]
+        [Fact(Skip = "Needs to be recorded again")]
```
Comment (maddieclayton) [inline src/ResourceManager/Insights/Commands.Insights.Test/ScenarioTests/ActionGroupsTests.cs]: Needs an issue filed against Insights

**Comment by scottpazelt** [comment]
I downloaded the latest 5.1.1 version of azurerm.insights from PSGallery and I'm still receiving the error below.

Get-AzureRmDiagnosticSetting : Operation returned an invalid status code 'NotFound'
At line:1 char:1
+ Get-AzureRmDiagnosticSetting -ResourceId "/subscriptions/f3816459-420 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmDiagnosticSetting], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Insights.Diagnostics.GetAzureRmDiagnosticSettingCommand

**Comment by acroftslalom** [comment]
@scottpazelt It looks like the v.5.1.1 that is "released to customers" is still old;  you can compare the cmdlet bits to the commits for this change and see that it's still using old code/referenced libraries.

I believe this gets officially released 8/28:

https://github.com/Azure/azure-powershell/milestone/54

It looks like it's available in CloudShell today however.

**Comment by acroftslalom** [comment]
FYI I was able to go to:

https://shell.azure.com

And install the module "AzureRM.NetCore" (not regular AzureRM, it does not work in Linux), and successfully run the "Set-AzureRmDiagnosticSetting" that was failing.

**Comment by scottpazelt** [comment]
Thanks @acroftslalom

**Pull request merged**